packages/earth-map/src/components/layers/component.tsx

```diff
@@ -106,61 +108,125 @@ class LayersComponent extends React.PureComponent<ILayers, ILayersState> {
+            )}
+          </>
+        }>
+        {(!selected || panelExpanded) && (
+          <>
+            {activeLayers.length > 0 && (
+              <div className="ng-section-background ng-position-relative ng-padding-medium-bottom ng-margin-bottom">
+                <div className="ng-space-between ng-padding-small-bottom ng-padding-medium-horizontal ng-padding-medium-top">
+                  <h2 className="ng-text-display-s ng-body-color ng-margin-remove">Selected Layers</h2>
+                  <a onClick={() => setLayersActive([])}>deselect all</a>
+                </div>
+                {activeLayers.map((layer) => {
+                  return (
+                    <ListItem
+                      active
+                      title={layer.name}
+                      key={`${layer.slug}-${layer.organization}`}
+                      onClick={debounce(() => this.onToggleLayer(layer), 200)}
+                      organization={(group.length > 1) && layer.organization}
+                      labels={layer.category} />
+                  )
+                })}
               </div>
-            ))}
-
-          <div className="ng-ep-border-top ng-margin-top ng-padding-medium-top">
-            <LayerComponent
-              layer={{ name: 'Labels', key: 'labels', slug: 'labels' }}
-              active={mapLabels}
-              toggleLayer={() => this.onLabels()}
-            />
-            <LayerComponent
-              layer={{ name: 'Roads', key: 'roads', slug: 'roads' }}
-              active={mapRoads}
-              toggleLayer={() => this.onRoads()}
-            />
-          </div>
-        </div>
-      </>
+            )}
+            <div className="ng-section-background ng-position-relative ng-padding-medium-bottom">
+              <h2 className="ng-padding-small-bottom ng-padding-medium-horizontal ng-padding-medium-top ng-text-display-s ng-body-color ng-margin-remove">Other</h2>
+              <ListItem
+                title="Labels"
+                active={mapLabels}
+                key="labels"
+                onClick={debounce(() => this.onLabels(), 200)} />
+              <ListItem
+                title="Roads"
+                active={mapRoads}
+                key="roads"
+                onClick={debounce(() => this.onRoads(), 200)} />
+            </div>
+            <InfiniteList
+              title= "Widget layers"
+              data={layers.results}
+              loading={loading}
+              nextPageCursor={nextPageCursor}
+              onNextPage={nextLayersPage}
+              pageSize={30}>
+              {(layer) => (
+                <ListItem
+                  hint={layer.$searchHint.name}
+                  title={layer.name}
+                  active={!!activeLayers.find((x) => x.slug === layer.slug)}
+                  key={`${layer.slug}-${layer.organization}`}
+                  onClick={debounce(() => this.onToggleLayer(layer), 200)}
```

Comment:
why do you need a debounce here?

Reply:
I remember there was an error inside Layer Manager when you toggle a layer too fast.